taichi/transforms/scalarize.cpp

```diff
@@ -0,0 +1,182 @@
+#include "taichi/ir/ir.h"
+#include "taichi/ir/statements.h"
+#include "taichi/ir/transforms.h"
+#include "taichi/ir/visitors.h"
+#include "taichi/system/profiler.h"
+
+TLANG_NAMESPACE_BEGIN
+class MatrixInitRemoval : public IRVisitor {
```

Comment:
Is `MatrixInitRemoval` not used in this PR? Shall we move it to the following PR which uses it?

Reply:
Sure, removed MatrixInitRemoval and left a TODO here